### DISCUSSION TITLE
https://starter.obytes.com/guides/authentication/

### DISCUSSION THREAD

**Discussion by giscus** [obytes/react-native-template-obytes#283] [Docs Comments]
# https://starter.obytes.com/guides/authentication/

How to manage authentication in your app.

https://starter.obytes.com/guides/authentication/

**Comment by ahmedShaheer3** [comment]
Zustand works very well with TypeScript and can be easily used outside React Three, which means more flexibility.    this would React Tree NOT React Three

**Reply by yjose** [reply]
fixed

**Comment by jvn4dev** [comment]
signIn: TThe function performs user sign-in.

a T should be removed.

**Comment by ibbybuilds** [comment]
I have a question regarding authentication flow handling. Shouldn't we control the navigation flow (i.e., directing authenticated users to the home stack and unauthenticated users to the auth stack) in the root _layout rather than (app)/_layout?

My concern is that with the current approach in (app)/_layout:

Users have momentary access to the (app)/_layout
We're unnecessarily mounting the home stack
Then performing an additional redirect back to the auth stack

Wouldn't handling this at the root layout level be more efficient to prevent unnecessary mounting and redirects?

**Reply by RakibulBh** [reply]
<Providers>
      <Stack>
        <Stack.Protected guard={status === 'signIn'}>
          <Stack.Screen name="(app)" options={{ headerShown: false }} />
        </Stack.Protected>
        <Stack.Protected guard={status !== 'signIn'}>
          <Stack.Screen name="login" options={{ headerShown: false }} />
        </Stack.Protected>
        <Stack.Screen name="onboarding" options={{ headerShown: false }} />
      </Stack>
    </Providers>

Here! Use the new protected routes in the new Expo Update

**Comment by RakibulBh** [comment]
Suggestion: How would you store user's info, e.g name, email to access within the app? how will it integrate?
